packages/sanity/schemas/product.ts

```diff
@@ -37,20 +38,20 @@ export default {
           to: [{ type: "category" }],
         },
       ],
-    },
-    {
+    }),
+    defineField({
       name: "images",
       title: "Images",
       type: "array",
       of: [
-        {
+        defineField({
           name: "productImage",
           title: "Product Image",
           type: "productImage",
-        },
+        }),
```

Comment:
Apologies for the timing conflict here, but @sarmeyer recently merged [this PR](https://github.com/FormidableLabs/nextjs-sanity-fe/pull/164) that made some changes to the way images are handled to be a bit more idiomatic. Can we cherry-pick those changes into this PR so we don't accidentally undo her work?

Reply:
Sure thing! ill take a look at her PR and amend my pr :)

Reply:
@sarmeyer ive updated this branch with the changes you added in #164, can you please review them to make sure ive not missed anything. 

thanks in advance :)